app/scripts/modules/core/src/account/AccountService.spec.ts

```diff
@@ -1,18 +1,26 @@
-import { mockHttpClient } from 'core/api/mock/jasmine';
 import { mock } from 'angular';
-import { $rootScope } from 'ngimport';
+import { mockHttpClient } from 'core/api/mock/jasmine';
+import { MockHttpClient } from 'core/api/mock/mockHttpClient';
 
 import { SETTINGS } from 'core/config/settings';
+import { $rootScope } from 'ngimport';
+import { CloudProviderRegistry } from '../cloudProvider';
 
 import { AccountService, IAccount } from './AccountService';
-import { CloudProviderRegistry } from '../cloudProvider';
+
+function flush<T>(http: MockHttpClient, promise: PromiseLike<T>): Promise<T> {
+  return http
+    .flush()
+    .then(() => setTimeout(() => $rootScope.$digest()))
+    .then(() => promise);
+}
```

Comment:
it forces another digest loop _in the next tick_ which kicks off the $q.when() chain in AccountService.  It basically jump starts AngularJS.  If it's done in the same tick, it becomes a noop because a digest was already happening (I assume, just making guesses here).